src/DynamoCoreWpf/ViewModels/Watch3D/DynamoGeometryModel3D.cs

```diff
@@ -13,24 +14,187 @@
+
+        private bool isSpecialData;
+        /// <summary>
+        /// Is this model marked as a special render package.
+        /// </summary>
+        public bool IsSpecialRenderPackageData { get => isSpecialData; internal set { SetAffectsRender(ref isSpecialData, value); } }
+
+        private bool hasTransparencyData;
+        /// <summary>
+        /// Does this model have alpha < 255.
+        /// </summary>
+        public bool HasTransparencyData { get => hasTransparencyData; internal set { SetAffectsRender(ref hasTransparencyData, value); } }
+
+        private bool requiresPerVertexColor;
+        /// <summary>
+        /// Should this model display vertex colors.
+        /// </summary>
+        public bool RequiresPerVertexColor { get => requiresPerVertexColor; internal set { SetAffectsRender(ref requiresPerVertexColor, value); } }
+
+        private bool isFlatShaded;
+        /// <summary>
+        /// Should this model disregard lighting calculations and display unlit texture or vertex colors.
+        /// </summary>
+        public bool IsFlatShaded { get => isFlatShaded; internal set { SetAffectsRender(ref isFlatShaded, value); } }
+
+        /// <summary>
+        /// Generates an int that packs all enum flags into a single int.
+        /// Can be decoded using binary &
+        /// ie - Flags & 1 = IsFrozen
+        ///  - if flags == 000000 - all flags are off
+        ///  - if flags == 000001 - frozen is enabled
+        ///  - if flags == 100001 - frozen and flatshade are enabled.
+        /// </summary>
+        /// <returns></returns>
+        private int GenerateEnumFromState()
+        {
+            var finalFlag = (int)(DynamoMeshShaderFlags.None)
+                 + (int)(IsFrozenData ? DynamoMeshShaderFlags.IsFrozen : 0)
+                 + (int)(IsSelectedData ? DynamoMeshShaderFlags.IsSelected : 0)
+                  + (int)(IsIsolatedData ? DynamoMeshShaderFlags.IsIsolated : 0)
+                   + (int)(IsSpecialRenderPackageData ? DynamoMeshShaderFlags.IsSpecialRenderPackage : 0)
+                    + (int)(HasTransparencyData ? DynamoMeshShaderFlags.HasTransparency : 0)
+                     + (int)(RequiresPerVertexColor ? DynamoMeshShaderFlags.RequiresPerVertexColor : 0)
+                      + (int)(IsFlatShaded ? DynamoMeshShaderFlags.FlatShade : 0);
+
+            return finalFlag;
+        }
+
+        protected override void OnUpdatePerModelStruct(RenderContext context)
+        {
+            base.OnUpdatePerModelStruct(context);
+            //store the entire state of all our flags in the X component and decode in the vertex shader.
+            //we'll pass this from the vertex shader to the fragment shader to determine color states.
+            //Params is a helix material builtin and maps to vParams in the shader.
+
+            //TODO this gets called whenever a render occurs, make sure it is fast.
+            modelStruct.Params.X = GenerateEnumFromState();
```

Comment:
here is where the data is actually passed to the struct that helix will send to the shader. Params is a builtin attribute that is passed to the vertexShader as `vParams` - we're just using the X component - nothing else is using it. `vParams` is around as a general structure for sending data to the shaders.

Reply:
The helix `ModelStruct` type also has a `Color` property, which I guess could be used to pass color data to the shader?

Reply:
I think that is available to pass a single color for the entire model, I believe in the shader it turns into `vColor`. This is defined in the common buffer for all meshes:https://github.com/helix-toolkit/helix-toolkit/blob/76dc523c9bf928ef2c6f6e72246df28959cc0336/Source/Examples/WPF.SharpDX/CustomShaderDemo/Shaders/CommonBuffers.hlsl#L56 
 I don't see this used in our old implementations anywhere.